backend/modules/cohorts/service.py

```
@@ -141,10 +141,36 @@ def get_cohorts(
+        cache_key = f"professor_cohorts:{user_id}:{user_role}:{skip}:{limit}:{search or 'none'}:{status or 'none'}"
+        
+        # Check Redis cache first
+        cache_check_start = time.time()
+        cached_result = redis_manager.get(cache_key)
+        cache_check_elapsed = time.time() - cache_check_start
+        
+        # #region agent log
+        try:
+            with open('/Users/amybihag/n-aible_edtech_sims/.cursor/debug.log', 'a') as f:
+                f.write(json_lib.dumps({"sessionId":"debug-session","runId":"run1","hypothesisId":"PROF_COHORTS","location":"cohorts/service.py:get_cohorts","message":"Cache check completed","data":{"cache_hit":cached_result is not None,"elapsed_ms":cache_check_elapsed*1000,"user_id":user_id},"timestamp":int(time.time()*1000)})+"\n")
+        except: pass
```

Ruff (0.14.8)

Do not use bare `except`

(E722)